sdk/resourcemanager/Azure.ResourceManager.Core/src/Resources/SubscriptionProviderIdentifier.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.ResourceManager.Core
+{
+    /// <summary>
+    /// The identifier of a Provider from Subscription.
+    /// </summary>
+    public class SubscriptionProviderIdentifier : TenantProviderIdentifier
```

Comment:
I think this should extend subscriptionResourceIdentifier - it is more important from the customer perspective that the SubscriptionId property is available than whether this is a provider id or not.

Reply:
There should still be a subscription Id.  This inheritance follows the "Properties shouldn't be nullable" and if we extend subscription then you won't know when the provider namespace is nullable without always having to check